app/controllers/v0/sessions_controller.rb

```diff
@@ -70,6 +70,14 @@ def saml_callback
 
     private
 
+    def auth_error_code(code)
+      if code == '005' && validate_session
+        UserSessionForm::ERRORS[:saml_replay_valid_session][:code]
+      else
+        code
+      end
+    end
```

Comment:
hmmm, i would liked to see this baked into the one of the underlying models rather than a sessions_controller method. Can it be handled within UserSessionForm or one of the SamlResponses?

Reply:
I don't think this belongs on UserSessionForm, if anything its a special case the SAML::Response::Login... if it has knowledge of an existing / valid session (one that is not expired), it should be handled there. UserSessionForm is only meant to be evaluated when SAML::Response::Login is VALID, in this case it is technically not valid.